components/fxa-client/src/lib.rs

```diff
@@ -162,33 +162,33 @@ impl FirefoxAccount {
     }
 
     /// Get the Sync Token Server endpoint URL.
-    pub fn get_token_server_endpoint_url(&self) -> Result<Url> {
-        self.state.config.token_server_endpoint_url()
+    pub fn get_token_server_endpoint_url(&self) -> Result<String> {
+        Ok(self.state.config.token_server_endpoint_url()?.into_string())
     }
 
     /// Get the pairing URL to navigate to on the Auth side (typically
     /// a computer).
-    pub fn get_pairing_authority_url(&self) -> Result<Url> {
+    pub fn get_pairing_authority_url(&self) -> Result<String> {
         // Special case for the production server, we use the shorter firefox.com/pair URL.
         if self.state.config.content_url()? == Url::parse(config::CONTENT_URL_RELEASE)? {
-            return Ok(Url::parse("https://firefox.com/pair")?);
+            return Ok("https://firefox.com/pair".to_owned());
         }
         // Similarly special case for the China server.
         if self.state.config.content_url()? == Url::parse(config::CONTENT_URL_CHINA)? {
-            return Ok(Url::parse("https://firefox.com.cn/pair")?);
+            return Ok("https://firefox.com.cn/pair".to_owned());
         }
-        Ok(self.state.config.pair_url()?)
+        Ok(self.state.config.pair_url()?.into_string())
     }
 
     /// Get the "connection succeeded" page URL.
     /// It is typically used to redirect the user after
     /// having intercepted the OAuth login-flow state/code
     /// redirection.
-    pub fn get_connection_success_url(&self) -> Result<Url> {
+    pub fn get_connection_success_url(&self) -> Result<String> {
```

Comment:
Oh, interesting, is this because we don't have support for declaring "extern" types in the IDL, so we can't declare it to return a `Url`?

Reply:
Yup indeed, for what it's worth the current FFI also calls `to_string()`.